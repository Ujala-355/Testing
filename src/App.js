
import './App.css';
import {useState} from "react";
function App() {
    const [data,setData]=useState("");
    const [text,setText]=useState("");
  return (
    <div className="App">
        <h1>First React Test case</h1>
        <h1>Test Click Event with Button</h1>
        <h1>Snapshot testing</h1>
        <br/>
        <img title="12 img papi 5675"  src="https://img.freepik.com/free-photo/puppy-that-is-walking-snow_1340-37228.jpg" alt="img 1"/>
        <br/>
        <input type="text" value={data} onChange={(e)=>setData(e.target.value+"test")}/>
        <br/>
        <button onClick={()=>setText("update data")}>Update Data</button>
        <h1>{text}</h1>
    </div>
  );
}
                         
export default App;
