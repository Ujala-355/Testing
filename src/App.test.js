import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

test("functional props testing", async()=>{
    const testFunction=jest.fn();
    render(<App testFunction={testFunction}/>)
    const btn=screen.getByRole("button");
    await userEvent.click(btn);
    expect(testFunction).toBeCalled();

})