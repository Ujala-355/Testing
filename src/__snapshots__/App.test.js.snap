// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for app component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <h1>
          First React Test case
        </h1>
        <h1>
          Test Click Event with Button
        </h1>
        <h1>
          Snapshot testing
        </h1>
        <br />
        <img
          alt="img 1"
          src="https://img.freepik.com/free-photo/puppy-that-is-walking-snow_1340-37228.jpg"
          title="12 img papi 5675"
        />
        <br />
        <input
          type="text"
          value=""
        />
        <br />
        <button>
          Update Data
        </button>
        <h1 />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <h1>
        First React Test case
      </h1>
      <h1>
        Test Click Event with Button
      </h1>
      <h1>
        Snapshot testing
      </h1>
      <br />
      <img
        alt="img 1"
        src="https://img.freepik.com/free-photo/puppy-that-is-walking-snow_1340-37228.jpg"
        title="12 img papi 5675"
      />
      <br />
      <input
        type="text"
        value=""
      />
      <br />
      <button>
        Update Data
      </button>
      <h1 />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
